dot/parachain/backing/active_leaves_update.go

```diff
@@ -168,64 +69,17 @@ func (cb *CandidateBacking) ProcessActiveLeavesUpdateSignal(update parachaintype
 	return nil
 }
 
-// updateCandidateSecondedAtDepth updates candidates seconded at depth under new active leaves.
-func updateCandidateSecondedAtDepth(
-	wg *sync.WaitGroup, mut *sync.Mutex, subSystemToOverseer chan<- any,
-	candidateHash parachaintypes.CandidateHash, paraID parachaintypes.ParaID,
-	leafHash common.Hash, secondedAtDepth map[parachaintypes.ParaID]*btree.Map[uint, parachaintypes.CandidateHash],
-) {
-	defer wg.Done()
-
-	getTreeMembership := parachaintypes.ProspectiveParachainsMessageGetTreeMembership{
-		ParaID:        paraID,
-		CandidateHash: candidateHash,
-		ResponseCh:    make(chan []parachaintypes.FragmentTreeMembership),
-	}
-
-	var membership []parachaintypes.FragmentTreeMembership
-
-	subSystemToOverseer <- getTreeMembership
-	select {
-	case membership = <-getTreeMembership.ResponseCh:
-	case <-time.After(parachaintypes.SubsystemRequestTimeout):
-		logger.Errorf("getting fragment tree membership: %w; candidate: %s, para-id: %d",
-			parachaintypes.ErrSubsystemRequestTimeout, candidateHash, paraID)
-		return
-	}
-
-	for _, m := range membership {
-		if m.RelayParent == leafHash {
-			mut.Lock()
-
-			tree, ok := secondedAtDepth[paraID]
-			if !ok {
-				tree = new(btree.Map[uint, parachaintypes.CandidateHash])
-			}
-
-			for _, depth := range m.Depths {
-				tree.Load(depth, candidateHash)
-			}
-			mut.Unlock()
-		}
-	}
-
-}
-
 func (cb *CandidateBacking) cleanUpPerRelayParentByLeafAncestry() {
-	remaining := make(map[common.Hash]bool)
-
-	for hash := range cb.perLeaf {
-		remaining[hash] = true
-	}
-
 	allowedRelayParents := cb.ImplicitView.AllAllowedRelayParents()
+
+	uniqueAllowedRelayParents := make(map[common.Hash]struct{})
```

Comment:
```suggestion
	uniqueAllowedRelayParents := make(map[common.Hash]any)
```
I usually just use `nil` when populating hash sets (empty value maps).

Reply:
I see your approach of using `nil` as the value in a `map[common.Hash]any`, which keeps the map structure simple. However, I personally prefer using `struct{}` for semantic clarity.

Using **struct{}** makes it explicitly clear that this map is being used as a set rather than implying that values might have significance.

That said, if there's a strong preference for `any` in our codebase, I'm happy to align with the convention! Let me know what you think.

Reply:
After looking into it, I found out that storing `any` uses way more memory than `struct{}`